### ISSUE TITLE
decoder & encoder overrides ignored when input/output value are of same type

### ISSUE THREAD

**Issue by sunbit** [lidatong/dataclasses-json#245]
Given this dataclass:

```
import html 

@dataclass_json(undefined=Undefined.RAISE)
@dataclass(frozen=True)
class Problem:
    id: str
    description: str = field(metadata=config(decoder=html.unescape))

```

where i want my dataclass to have the string unescaped. The problem with that is that custom encoder/decoder overrides are completely ignored when resulting value is of same type of input value due to this code [here](https://github.com/lidatong/dataclasses-json/blob/master/dataclasses_json/core.py#L181)

```
        if (field.name in overrides
                and overrides[field.name].decoder is not None):
            # FIXME hack
            if field_type is type(field_value):
                init_kwargs[field.name] = field_value
            else:
                init_kwargs[field.name] = overrides[field.name].decoder(
                    field_value)
```

I'm not sure what exactly this code is trying to avoid by ignoring the decoders at this point, but the FIXME comment suggests me that this needs some thought. IMO it's is completely valid to decode/encode to the same type, another example would be encode/decode to different character encodings.

So the question is it why are decoders/encoders deleted in this case?

**Comment by RevanthRameshkumar** [comment]
Yes, this is a problem for me as well! Due to wanting to change character encodings and other string ops where the encoder/decoder is str <-> str
